File: Week - 1/Assignment - 1/LinearSearch.cpp
#include <iostream>
using namespace std;

int linearSearch(int arr[], int n, int target) {
    for (int i = 0; i < n; i++) {
        if (arr[i] == target) {
            return i;
        }
    }

    return -1;
}

int main() {
    int n;

    cout << "Enter the number of elements: ";
    cin >> n;

    int arr[n];

    cout << "Enter the elements:\n";
    for (int i = 0; i < n; i++) {
        cin >> arr[i];
    }

    int target;
    cout << "Enter the element to search: ";
    cin >> target;

    int result = linearSearch(arr, n, target);

    if (result != -1) {
        cout << "Element " << target
             << " found at index " << result << "." << endl;
    } else {
        cout << "Element " << target
             << " not found in the array." << endl;
    }

    return 0;
}

File: Week - 1/Assignment - 1/LinearSearch.py
def linear_search(arr, target):
    for i in range(len(arr)):
        if arr[i] == target:
            return i
    return -1

n = int(input("Enter the number of elements: "))

arr = []
print("Enter the elements:")

for i in range(n):
    element = int(input(f"Element {i + 1}: "))
    arr.append(element)

target = int(input("Enter the element to search: "))

result = linear_search(arr, target)

if result != -1:
    print(f"Element {target} found at index {result}.")
else:
    print(f"Element {target} not found in the array.")